<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1">


<meta name="date" content="2019-10-20" />

<title>A short introduction to the corto Package</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">A short introduction to the <em>corto</em> Package</h1>
<h4 class="author"><div class="line-block">Federico M. Giorgi<br />
Department of Pharmacy and Biotechnology, University of Bologna</div></h4>
<h4 class="date">20 October 2019</h4>



<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>The <em>corto</em> (“correlation tool”) package provides a pipeline to infer networks between “centroid” and “target” variables in a dataset, using a combination of Spearman correlation and Data Processing Inequality (DPI), first proposed in [1]. The main application of <em>corto</em> is in the field of Bioinformatics and Transcriptomics, where co-occurrence between variables can be used as a mean to infer regulatory mechanisms [2] or gene functions [3]. In this field, usually the tested features are genes (or rather, their expression profile across samples), whereas the centroids are Transcription Factors (TFs) and their targets are Target Genes (TGs). The TF-TG co-expression can hint at a causal regulatory relationship, as proven in many studies [4,5,6]. The <em>corto</em> tool replicates the well-established pipeline of the ARACNe family of tools [7,8,9]</p>
<p>In brief, <em>corto</em> operates using the following steps:</p>
<ol style="list-style-type: decimal">
<li>Load an gene expression matrix and a list of user-provided centroids (e.g. TFs). In the case of RNA-Seq, VST-normalized data, as described in [10].</li>
<li>Calculate all pairwise Spearman correlation coefficients between centroid and target features. The rank transformation operated by the Spearman correlation coefficient is a common procedure used in co-expression based studies, due to its benefits in reducing the effects of outliers [11].</li>
<li>Filter out all centroid-target features whose correlation coefficient <em>absolute value</em> is below the provided p-value threshold <em>p</em>.</li>
<li>Apply DPI to all centroid-centroid-target triplets, in order to identify which centroid-target correlation is stronger and identify the most likely association (e.g. which TF is the most likely regulator of the TG in the dataset).</li>
<li>All edges are tested for Data Processing Inequality in a number of bootstrapped versions of the same input matrix (specified by the <em>nbootstraps</em> parameter, 100 by default, as in ARACNE-AP [8]). This step can be run in parallel by specifiying the number of threads using the <em>nthreads</em> parameter. The number of occurrences of each edge (if they survived DPI) is calculated. This number can range from 0 (the edge is not significant in any bootstrap) to <em>nbootstraps</em>+1 (the edge is significant in all bootstraps, and also in the original matrix).</li>
<li>A network is generated in the form of a list, where each element is a centroid-based list of targets. In order to follow the structure of the <em>regulon</em> class implemented by downstream analysis packages (e.g. VIPER [12]), each target is characterized by two parameters:
<ul>
<li>The <em>tfmode</em>, the mode of action, i.e. the Spearman correlation coefficient between the target and the centroid in the original input matrix</li>
<li>The <em>likelihood</em> of the interaction, calculated as the number of bootstraps in which the edge survives DPI, divided by the total number of bootstraps (in this impelmentation, the presence of an edge in the original, non-bootstrapped matrix is considered as an additional evidence)</li>
</ul></li>
</ol>
</div>
<div id="running-corto" class="section level1">
<h1>Running <em>corto</em></h1>
<p>Here is how to run <em>corto</em>. First, install the package:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">install.packages</span>(<span class="st">&quot;corto&quot;</span>)</a></code></pre></div>
<p>Then, load it:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw">library</span>(corto)</a></code></pre></div>
<p>Then, you can see how the input matrix looks like. For example, this dataset comes from the TCGA mesothelioma project [13] and measures the expression of 10021 genes across 87 samples:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"><span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;inmat.rda&quot;</span>,<span class="dt">package=</span><span class="st">&quot;corto&quot;</span>))</a>
<a class="sourceLine" id="cb3-2" title="2">inmat[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a>
<a class="sourceLine" id="cb3-3" title="3"><span class="co">#         TCGA-3H-AB3K-01A TCGA-3H-AB3L-01A TCGA-3H-AB3M-01A</span></a>
<a class="sourceLine" id="cb3-4" title="4"><span class="co"># ZSCAN29         9.710627        10.045033         9.581725</span></a>
<a class="sourceLine" id="cb3-5" title="5"><span class="co"># MYOG            5.699747         5.699747         6.044523</span></a>
<a class="sourceLine" id="cb3-6" title="6"><span class="co"># PAX3            5.926491         5.699747         6.318039</span></a>
<a class="sourceLine" id="cb3-7" title="7"><span class="co"># BNC1           13.059897        13.289953        13.932597</span></a>
<a class="sourceLine" id="cb3-8" title="8"><span class="co"># SPIB            7.581019         7.818206         6.780734</span></a>
<a class="sourceLine" id="cb3-9" title="9"><span class="co">#         TCGA-3H-AB3O-01A TCGA-3H-AB3S-01A</span></a>
<a class="sourceLine" id="cb3-10" title="10"><span class="co"># ZSCAN29         9.966440         9.317134</span></a>
<a class="sourceLine" id="cb3-11" title="11"><span class="co"># MYOG            5.699747         6.335294</span></a>
<a class="sourceLine" id="cb3-12" title="12"><span class="co"># PAX3            6.667444         6.303159</span></a>
<a class="sourceLine" id="cb3-13" title="13"><span class="co"># BNC1           11.860305        13.143449</span></a>
<a class="sourceLine" id="cb3-14" title="14"><span class="co"># SPIB            7.629027         5.985807</span></a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="kw">dim</span>(inmat)</a>
<a class="sourceLine" id="cb4-2" title="2"><span class="co"># [1] 7000   87</span></a></code></pre></div>
<p>Another input needed by <em>corto</em> is a list of centroid features. In our case, we can specify a list of TFs generated from Gene Ontology with the term “Regulation of Transcription” [14].</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1"><span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;centroids.rda&quot;</span>,<span class="dt">package=</span><span class="st">&quot;corto&quot;</span>))</a>
<a class="sourceLine" id="cb5-2" title="2">centroids[<span class="dv">15</span>]</a>
<a class="sourceLine" id="cb5-3" title="3"><span class="co"># [1] &quot;BNC1&quot;</span></a></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1"><span class="kw">length</span>(centroids)</a>
<a class="sourceLine" id="cb6-2" title="2"><span class="co"># [1] 294</span></a></code></pre></div>
<p>Finally, we can run <em>corto</em>. In this example, we will run it with p-value threshold of 1e-30, 10 bootstraps and 2 threads</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1">regulon&lt;-<span class="kw">corto</span>(inmat,<span class="dt">centroids=</span>centroids,<span class="dt">nbootstraps=</span><span class="dv">10</span>,<span class="dt">p=</span><span class="fl">1e-30</span>,<span class="dt">nthreads=</span><span class="dv">2</span>)</a>
<a class="sourceLine" id="cb7-2" title="2"><span class="co"># Input Matrix has 87 samples and 10021 features</span></a>
<a class="sourceLine" id="cb7-3" title="3"><span class="co"># Correlation Coefficient Threshold is: 0.889962633618839</span></a>
<a class="sourceLine" id="cb7-4" title="4"><span class="co"># Removed 112 features with zero variance</span></a>
<a class="sourceLine" id="cb7-5" title="5"><span class="co"># Calculating pairwise correlations</span></a>
<a class="sourceLine" id="cb7-6" title="6"><span class="co"># Initial testing of triplets for DPI</span></a>
<a class="sourceLine" id="cb7-7" title="7"><span class="co"># 246 edges passed the initial threshold</span></a>
<a class="sourceLine" id="cb7-8" title="8"><span class="co"># Building DPI network from 37 centroids and 136 targets</span></a>
<a class="sourceLine" id="cb7-9" title="9"><span class="co"># Running 100 bootstraps with 2 thread(s)</span></a>
<a class="sourceLine" id="cb7-10" title="10"><span class="co"># Calculating edge likelihood</span></a>
<a class="sourceLine" id="cb7-11" title="11"><span class="co"># Generating regulon object</span></a></code></pre></div>
<p>The regulon object is a list:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1">regulon[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a>
<a class="sourceLine" id="cb8-2" title="2"><span class="co"># $SCRT1</span></a>
<a class="sourceLine" id="cb8-3" title="3"><span class="co"># $SCRT1$tfmode</span></a>
<a class="sourceLine" id="cb8-4" title="4"><span class="co">#    ACTL6B    CACNG2     CDY1B </span></a>
<a class="sourceLine" id="cb8-5" title="5"><span class="co"># 0.9032878 0.8994689 0.9132632 </span></a>
<a class="sourceLine" id="cb8-6" title="6"><span class="co"># </span></a>
<a class="sourceLine" id="cb8-7" title="7"><span class="co"># $SCRT1$likelihood</span></a>
<a class="sourceLine" id="cb8-8" title="8"><span class="co"># [1] 0.2727273 0.3636364 0.3636364</span></a>
<a class="sourceLine" id="cb8-9" title="9"><span class="co"># </span></a>
<a class="sourceLine" id="cb8-10" title="10"><span class="co"># </span></a>
<a class="sourceLine" id="cb8-11" title="11"><span class="co"># $ELF5</span></a>
<a class="sourceLine" id="cb8-12" title="12"><span class="co"># $ELF5$tfmode</span></a>
<a class="sourceLine" id="cb8-13" title="13"><span class="co">#      APOH  ARHGEF38       C8B   CEACAM5    CLDN18      CPB2    CXCL17 </span></a>
<a class="sourceLine" id="cb8-14" title="14"><span class="co"># 0.9018898 0.9079885 0.9054840 0.9150803 0.9353695 0.9221696 0.8943170 </span></a>
<a class="sourceLine" id="cb8-15" title="15"><span class="co">#       FGA </span></a>
<a class="sourceLine" id="cb8-16" title="16"><span class="co"># 0.9042021 </span></a>
<a class="sourceLine" id="cb8-17" title="17"><span class="co"># </span></a>
<a class="sourceLine" id="cb8-18" title="18"><span class="co"># $ELF5$likelihood</span></a>
<a class="sourceLine" id="cb8-19" title="19"><span class="co"># [1] 0.7272727 0.7272727 0.7272727 0.7272727 0.8181818 0.8181818 0.1818182</span></a>
<a class="sourceLine" id="cb8-20" title="20"><span class="co"># [8] 0.7272727</span></a></code></pre></div>
<p>The regulon in this dataset is composed of 34 final centroids with at least one target:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1"><span class="kw">length</span>(regulon)</a>
<a class="sourceLine" id="cb9-2" title="2"><span class="co"># [1] 28</span></a></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1"><span class="kw">names</span>(regulon)</a>
<a class="sourceLine" id="cb10-2" title="2"><span class="co">#  [1] &quot;SCRT1&quot;   &quot;ELF5&quot;    &quot;NEUROG1&quot; &quot;ZSCAN29&quot; &quot;BCL6B&quot;   &quot;MYOG&quot;    &quot;REST&quot;   </span></a>
<a class="sourceLine" id="cb10-3" title="3"><span class="co">#  [8] &quot;MYBL2&quot;   &quot;FEZF2&quot;   &quot;PAX5&quot;    &quot;UHRF1&quot;   &quot;DMBX1&quot;   &quot;SOX7&quot;    &quot;PAX3&quot;   </span></a>
<a class="sourceLine" id="cb10-4" title="4"><span class="co"># [15] &quot;BNC1&quot;    &quot;TBX21&quot;   &quot;IKZF3&quot;   &quot;EOMES&quot;   &quot;TFEC&quot;    &quot;ZNF831&quot;  &quot;FOXM1&quot;  </span></a>
<a class="sourceLine" id="cb10-5" title="5"><span class="co"># [22] &quot;NKX2-1&quot;  &quot;SPIB&quot;    &quot;RFX4&quot;    &quot;SPI1&quot;    &quot;NOBOX&quot;   &quot;FOS&quot;     &quot;FOXO3&quot;</span></a></code></pre></div>
</div>
<div id="correcting-with-cnv-data" class="section level1">
<h1>Correcting with CNV data</h1>
<p>As an additional, optional feature, <strong>corto</strong> gives the user the possibility to provide Copy Number Variation (CNV) data, if available, which can generate spurious correlations between TFs and targets [15]. The gene expression profiles for the target genes are corrected via linear regression and the residuals of the expression~cnv model substitute the original gene expression profile.</p>
<p>In this example, a CNV matrix is provided. The analysis will be run only for the features (rows) and samples (columns) present in both matrices</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1"><span class="kw">load</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>,<span class="st">&quot;cnvmat.rda&quot;</span>,<span class="dt">package=</span><span class="st">&quot;corto&quot;</span>,<span class="dt">mustWork=</span><span class="ot">TRUE</span>))</a>
<a class="sourceLine" id="cb11-2" title="2">regulon &lt;-<span class="st"> </span><span class="kw">corto</span>(inmat,<span class="dt">centroids=</span>centroids,<span class="dt">nthreads=</span><span class="dv">2</span>,<span class="dt">nbootstraps=</span><span class="dv">10</span>,<span class="dt">verbose=</span><span class="ot">TRUE</span>,<span class="dt">cnvmat=</span>cnvmat,<span class="dt">p=</span><span class="fl">0.01</span>)</a></code></pre></div>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<p>[1] Reverter, Antonio, and Eva KF Chan. “Combining partial correlation and an information theory approach to the reversed engineering of gene co-expression networks.” Bioinformatics 24.21 (2008): 2491-2497.</p>
<p>[2] D’Haeseleer, Patrik, Shoudan Liang, and Roland Somogyi. “Genetic network inference: from co-expression clustering to reverse engineering.” Bioinformatics 16.8 (2000): 707-726.</p>
<p>[3] Hansen, Bjoern O., et al. “Elucidating gene function and function evolution through comparison of co-expression networks of plants.” Frontiers in plant science 5 (2014): 394.</p>
<p>[4] Basso, Katia, et al. “Reverse engineering of regulatory networks in human B cells.” Nature genetics 37.4 (2005): 382. [5] Amar, David, Hershel Safer, and Ron Shamir. “Dissection of regulatory networks that are altered in disease via differential co-expression.” PLoS computational biology 9.3 (2013): e1002955.</p>
<p>[6] Vandepoele, Klaas, et al. “Unraveling transcriptional control in Arabidopsis using cis-regulatory elements and coexpression networks.” Plant physiology 150.2 (2009): 535-546.</p>
<p>[7] Margolin, Adam A., et al. “ARACNE: an algorithm for the reconstruction of gene regulatory networks in a mammalian cellular context.” BMC bioinformatics. Vol. 7. No. 1. BioMed Central, 2006.</p>
<p>[8] Lachmann, Alexander, et al. “ARACNe-AP: gene network reverse engineering through adaptive partitioning inference of mutual information.” Bioinformatics 32.14 (2016): 2233-2235.</p>
<p>[9] Khatamian, Alireza, et al. “SJARACNe: a scalable software tool for gene network reverse engineering from big data.” Bioinformatics 35.12 (2018): 2165-2166.</p>
<p>[10] Giorgi, Federico M., et al. “Comparative study of RNA-seq-and microarray-derived coexpression networks in Arabidopsis thaliana.” Bioinformatics 29.6 (2013): 717-724.</p>
<p>[11] Usadel, Björn, et al. “Co‐expression tools for plant biology: opportunities for hypothesis generation and caveats.” Plant, cell &amp; environment 32.12 (2009): 1633-1651.</p>
<p>[12] Alvarez, Mariano J., Federico Giorgi, and Andrea Califano. “Using viper, a package for Virtual Inference of Protein-activity by Enriched Regulon analysis.” Bioconductor (2014): 1-14.</p>
<p>[13] Ladanyi, Marc, et al. “The TCGA malignant pleural mesothelioma (MPM) project: VISTA expression and delineation of a novel clinical-molecular subtype of MPM.” (2018): 8516-8516.</p>
<p>[14] Gene Ontology Consortium. “The Gene Ontology (GO) database and informatics resource.” Nucleic acids research 32.suppl_1 (2004): D258-D261.</p>
<p>[15] Schubert, Michael, et al. “Gene networks in cancer are biased by aneuploidies and sample impurities.” BioRxiv (2019): 752816.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
